File: tests/unit/php/Model/CountrySubdivisionTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use InvalidArgumentException;
use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\CountrySubdivision;

class CountrySubdivisionTest extends TestCase
{
    private CountrySubdivision $countrySubdivision;

    public function setUp(): void
    {
        $this->countrySubdivision = new CountrySubdivision([]);
    }

    public function testPropertyName()
    {
        $this->countrySubdivision->setName('British Columbia');
        $this->assertEquals('British Columbia', $this->countrySubdivision->getName());

        try {
            $this->countrySubdivision->setName(str_repeat('a', 101));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyCode()
    {
        $this->countrySubdivision->setCode('CA-BC');
        $this->assertEquals('CA-BC', $this->countrySubdivision->getCode());

        try {
            $this->countrySubdivision->setCode(str_repeat('a', 101));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyParentCode()
    {
        $this->countrySubdivision->setParentCode('CA');
        $this->assertEquals('CA', $this->countrySubdivision->getParentCode());

        try {
            $this->countrySubdivision->setParentCode(str_repeat('a', 6));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }
}

File: tests/unit/java/com/trulioo/sdk/model/CountrySubdivisionTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

public class CountrySubdivisionTest {
    private CountrySubdivision countrySubdivision;

    @BeforeEach
    public void beforeEach() {
        countrySubdivision = new CountrySubdivision();
    }

    @Test
    public void testName() {
        String name = "British Columbia";
        countrySubdivision.setName(name);

        assertEquals(name, countrySubdivision.getName());
        assertEquals(name, new CountrySubdivision().name(name).getName());
    }

    @Test
    public void testCode() {
        String code = "BC";
        countrySubdivision.setCode(code);

        assertEquals(code, countrySubdivision.getCode());
        assertEquals(code, new CountrySubdivision().code(code).getCode());
    }

    @Test
    public void testParentCode() {
        String parentCode = "CA-BC";
        countrySubdivision.setParentCode(parentCode);

        assertEquals(parentCode, countrySubdivision.getParentCode());
        assertEquals(parentCode, new CountrySubdivision().parentCode(parentCode).getParentCode());
    }

    @Test
    public void testEquals() {
        String name = "British Columbia";
        CountrySubdivision countrySubdivision1 = new CountrySubdivision().name(name);

        assertEquals(countrySubdivision1, countrySubdivision1);
        assertEquals(countrySubdivision1, new CountrySubdivision().name(name));
        assertNotEquals(countrySubdivision1, new CountrySubdivision().name(name + "1"));
        assertNotEquals(countrySubdivision1, null);
    }

    @Test
    public void testHashcode() {
        String name = "British Columbia";

        assertEquals(new CountrySubdivision().name(name).hashCode(), new CountrySubdivision().name(name).hashCode());
    }

    @Test
    public void testToString() {
        String name = "British Columbia";

        assertEquals(new CountrySubdivision().name(name).toString(), new CountrySubdivision().name(name).toString());
    }
}
